src/ui/actions/app.ts

```diff
@@ -53,6 +58,107 @@ function now(): number {
+    // That's just the way that the Console.findMessages protocol command works.
+    // In that case (without overflow), like I said above, we never need to run
+    // any of this code. If we have overflow, and focus on, let's say the first
+    // 30 seconds of the recording, and we *don't* get overflow on the first
+    // thirty seconds, then we know that if the user were to change the window
+    // to the first 15 seconds, we don't need to refetch, and we also should not
+    // update the "most recently loaded" boundaries in the store, because that
+    // way the user could then focus on, for example, seconds 10 to 20 *also
+    // without refetching*.
+    // BTW - we don't have to figure out *all* of this right now :)
+    dispatch(clearMessages());
+
+    const sessionEndpoint = await sendMessage("Session.getEndpoint", {}, ThreadFront.sessionId!);
+
+    let beginPoint: string | null = null;
+    let endPoint: string | null = null;
+
+    if (sessionEndpoint.endpoint.time === focusRegion.endTime) {
+      const pointNearBeginning = await sendMessage(
+        "Session.getPointNearTime",
+        {
+          time: startTime,
+        },
+        ThreadFront.sessionId!
+      );
+
+      beginPoint = pointNearBeginning.point.point;
+      endPoint = sessionEndpoint.endpoint.point;
+    } else {
+      const [pointNearBeginning, pointNearEnd] = await Promise.all([
+        sendMessage(
+          "Session.getPointNearTime",
+          {
+            time: startTime,
+          },
+          ThreadFront.sessionId!
+        ),
+        sendMessage("Session.getPointNearTime", { time: endTime }, ThreadFront.sessionId!),
+      ]);
+
+      beginPoint = pointNearBeginning.point.point;
+      endPoint = pointNearEnd.point.point;
+    }
+
+    // TODO [bvaughn] Store overflow somewhere too (for soft focus)
+    // @ts-ignore TypeScript doesn't (yet) know about this return value.
+    const { messages, overflow } = await sendMessage(
+      "Console.findMessagesInRange",
+      {
+        range: { begin: beginPoint, end: endPoint },
+      },
+      ThreadFront.sessionId!
+    );
+
+    // Copied from ThreadFront.findConsoleMessages():
+    // TODO [bvaughn] Would be nice if this shared code with ThreadFront.
+    messages.forEach(message => {
+      const wiredMessage = message as WiredMessage;
+
+      const pause = new Pause(ThreadFront);
+      pause.instantiate(
+        message.pauseId,
+        message.point.point,
+        message.point.time,
+        !!message.point.frame,
+        message.data
+      );
+
+      if (message.argumentValues) {
+        wiredMessage.argumentValues = message.argumentValues.map(
+          value => new ValueFront(pause, value)
+        );
+      }
+
+      if (message.sourceId) {
+        message.sourceId = ThreadFront.getCorrespondingSourceIds(message.sourceId)[0];
+      }
+
+      dispatch(onConsoleMessage(wiredMessage));
+    });
```

Comment:
This feels pretty clunky. Would be nice to DRY this sort of stuff up _at least_ as it would be easy to _forget_ to do one of these things (as I did at one point) which would lead to errors.